pkg/api/steve/supportconfigs/generator.go

```diff
@@ -24,33 +25,87 @@ import (
 
 const (
 	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
-	Endpoint            = "/v1/generateSUSERancherSupportConfig"
+	Endpoint = "/v1/generateSUSERancherSupportConfig"
+	// NOTE: the name of the configmap will be the same for both Bring-Your-Own (BOY) and
+	// Pay-As-You-Go (PAYG) offerings. However, they will be in different namespaces to avoid
+	// clashing.
 	cspAdapterConfigmap = "csp-config"
 	tarContentType      = "application/x-tar"
 	logPrefix           = "support-config-generator"
+	// Pay-As-You-Go offering
+	PAYG = "PAYG"
+	// Bring-Your-Own offering
+	BYO = "BYO"
 )
 
 var errNotFound = errors.New("not implemented")
 
+var (
+	configMaps   v1.ConfigMapInterface
+	cspNamespace string
+)
+
 // Handler implements http.Handler - and serves supportconfigs (tar file which contains support relevant information)
 type Handler struct {
-	ConfigMaps           v1.ConfigMapInterface
+	scaledContext        *config.ScaledContext
 	SubjectAccessReviews authv1.SubjectAccessReviewInterface
 	adapterUtil          *cspadapter.ChartUtil
+	billingAdapterUtil   *cspbillingadapter.ChartUtil
 }
 
 // NewHandler creates a handler using the clients defined in scaledContext
 func NewHandler(scaledContext *config.ScaledContext) Handler {
 	return Handler{
-		ConfigMaps:           scaledContext.Core.ConfigMaps(cspadapter.ChartNamespace),
+		scaledContext:        scaledContext,
```

Comment:
Rather than removing this initialization, you can use `metav1.NamespaceAll` to configure the configmap interface to use all namespaces, and then use `GetNamespaced` instead of `Get` to provide the namespace that you want to retrieve the value from. This would allow you to keep the earlier initialization structure and still get a configmap from a potentially different namespace. You can see something similar in [metrics.go](https://github.com/rancher/rancher/blob/18ad6028a05d7745d11f47b50ece4fbec1a8997b/pkg/metrics/metrics.go#L156). and in [log.go](https://github.com/rancher/rancher/blob/18ad6028a05d7745d11f47b50ece4fbec1a8997b/pkg/clusterprovisioninglogger/log.go#L55).

Reply:
Thanks for the suggestion. That's a much better approach. I will make the change.